File: app/src/main/java/com/example/appregistro/DescriptionActivity.java
package com.example.appregistro;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.Button;
import android.widget.EditText;

public class DescriptionActivity extends AppCompatActivity {

    private EditText editDescripcion;
    private Button btnGuardarDescripcion;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_description);

        editDescripcion = findViewById(R.id.editDescripcion);
        btnGuardarDescripcion = findViewById(R.id.btnGuardarDescripcion);

        btnGuardarDescripcion.setOnClickListener(v -> {
            String descripcion = editDescripcion.getText().toString();
            Intent resultIntent = new Intent();
            resultIntent.putExtra("descripcion", descripcion);
            setResult(RESULT_OK, resultIntent);
            finish();
        });
    }
}

File: app/src/main/java/com/example/appregistro/ui/home/DescriptionActivity.kt
package com.example.appregistro.ui.home

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.appregistro.databinding.ActivityDescriptionBinding

class DescriptionActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDescriptionBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDescriptionBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val nombreGasto = intent.getStringExtra("nombreGasto")
        val costoGasto = intent.getStringExtra("costoGasto")

        binding.textDescripcion.text = "Gasto: $nombreGasto\nCosto: $costoGasto"
    }
}
